src/main/java/Calculator.java

```diff
@@ -1,72 +1,171 @@
+     */
     public Calculator multiply(MoneyValue other) {
-        if(!validateMoneyValue(other))
+        if (isMoneyValueInvalid(other)) {
             throw new MoneyValue.InvalidMoneyValueException(MoneyValue.INVALID_MONEY_VALUE_AS_STRING);
+        }
 
         MoneyValue otherInSameCurrency = mv_.getCurrency() == other.getCurrency() ?
                 other : cv_.convertTo(other, mv_.getCurrency());
         mv_ = mv_.multiply(otherInSameCurrency);
         return this;
     }
 
+    /**
+     * Subtracts the specified MoneyValue from the current MoneyValue in this calculator.
+     * If the currencies of the two MoneyValues are different, the specified MoneyValue
+     * is converted to the currency of the current MoneyValue using the Converter client.
+     * If the specified MoneyValue is invalid, an InvalidMoneyValueException is thrown.
+     *
+     * @param other the MoneyValue to subtract from the current MoneyValue
+     * @return this Calculator instance, updated with the new difference
+     * @throws MoneyValue.InvalidMoneyValueException if the specified MoneyValue is invalid
+     */
     public Calculator subtract(MoneyValue other) {
-        if(!validateMoneyValue(other))
+        if (isMoneyValueInvalid(other)) {
             throw new MoneyValue.InvalidMoneyValueException(MoneyValue.INVALID_MONEY_VALUE_AS_STRING);
+        }
 
         MoneyValue otherInSameCurrency = mv_.getCurrency() == other.getCurrency() ?
                 other : cv_.convertTo(other, mv_.getCurrency());
         mv_ = mv_.subtract(otherInSameCurrency);
         return this;
     }
 
+    /**
+     * Divides the current MoneyValue in this calculator by the specified MoneyValue.
+     * If the currencies of the two MoneyValues are different, the specified MoneyValue
+     * is converted to the currency of the current MoneyValue using the Converter client.
+     * If the specified MoneyValue is invalid or zero, an InvalidMoneyValueException is thrown.
+     *
+     * @param other the MoneyValue to divide by
+     * @return this Calculator instance, updated with the new quotient
+     * @throws MoneyValue.InvalidMoneyValueException if the specified MoneyValue is invalid or zero
+     */
     public Calculator divide(MoneyValue other) {
-        if(!validateMoneyValue(other) || other.getAmount().equals(BigDecimal.valueOf(0.0)))
+        if (isMoneyValueInvalid(other) || other.getAmount().equals(BigDecimal.valueOf(0.0))) {
             throw new MoneyValue.InvalidMoneyValueException(MoneyValue.INVALID_MONEY_VALUE_AS_STRING);
+        }
 
         MoneyValue otherInSameCurrency = mv_.getCurrency() == other.getCurrency() ?
                 other : cv_.convertTo(other, mv_.getCurrency());
         mv_ = mv_.divide(otherInSameCurrency);
         return this;
     }
 
-    private boolean validateMoneyValue(MoneyValue mv) {
-        if(mv == null)
-            return false;
-        return true;
+    /**
```

Comment:
The fact that it will get extended or could be is not something that should be mentioned here.